src/ray/core_worker/transport/direct_task_transport.cc

```diff
@@ -6,20 +6,26 @@ namespace ray {
 
 Status CoreWorkerDirectTaskSubmitter::SubmitTask(TaskSpecification task_spec) {
   resolver_.ResolveDependencies(task_spec, [this, task_spec]() {
-    // TODO(ekl) should have a queue per distinct resource type required
     absl::MutexLock lock(&mu_);
-    queued_tasks_.push_back(task_spec);
-    RequestNewWorkerIfNeeded(task_spec);
+    const TaskQueueKey queue_key(task_spec.GetSchedulingClass(),
+                                 task_spec.GetDependencyHash());
```

Comment:
I don't think we need to optimize this path so much that we keep a hash here, it should be rare to have plasma deps for direct calls. Consider just using the full dep vector here.